compiler/src/dotty/tools/io/ClassPath.scala

```diff
@@ -242,8 +242,8 @@ abstract class ClassPath {
       case Some((pkg, rest)) =>
         val rep = packages find (_.name == pkg) flatMap (_ findClass rest)
         rep map {
-          case x: ClassRep  => x
-          case x            => throw new FatalError("Unexpected ClassRep '%s' found searching for name '%s'".format(x, name))
+          case x: AnyClassRep => x
+          case x              => throw new FatalError("Unexpected ClassRep '%s' found searching for name '%s'".format(x, name))
```

Comment:
`rep` is an `AnyClassRep` so this case will never happen. I believe the original code intended to check the outer pointer, though I have no idea why, here's the commit that introduced the check: https://github.com/scala/scala/commit/a1a6ab90ac5fe7d5db2f6fba788ce14c1613d9c6